checkmarx-ltd/VisualStudio
# Plug 27

**Description**
By default display only 100 project and given option to search by project name and show all projects. Also Added one node in  CxVSPlugin.conf file BindProjectCount = -1

**Test Cases**
Test Case 1
- Search By project name is empty  && Show all projects unchecked && BindProjectCount == -1 
  Expected - Project <= 100 
  
Test Case 2
- Search By project name is empty  && Show all projects checked && BindProjectCount == -1
  Expected - display all projects
  
Test Case 3
- Search By project name contains value  && Show all projects unchecked && BindProjectCount == -1
  Expected - Find project list by name and  display searched Project list  <= 100  
  
Test Case 4
- Search By project name contains value  && Show all projects checked && BindProjectCount == -1
  Expected - Find project list by project name and display all searched Project list   
  
Test Case 5
- Search By project name contains value  && Show all projects checked && BindProjectCount == -1 and click on close/remove icon
  Expected - display all Project list  
  
Test Case 6
- Search By project name contains value  && Show all projects unchecked && BindProjectCount == -1 and click on close/remove icon
  Expected - Project <= 100 
  
Test Case 7
- BindProjectCount > -1 && Search By project name is empty
  Expected -  Show all projects should be hidden and display projects <= BindProjectCount 

Test Case 8
- BindProjectCount > -1 && Search By project name contains value
  Expected -  Show all projects should be hidden and  Find project list by project name and display searched Project list <= BindProjectCount

Test Case 9
- BindProjectCount > -1 && Search By project name contains value and click on close/remove icon
  Expected -  Project <= BindProjectCount
